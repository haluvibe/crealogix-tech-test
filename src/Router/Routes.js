// Root.js
import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import Main from "../Pages/Main/Main";
import History from "../Pages/History/History";
import NotFound from "../Pages/NotFound/NotFound";

export const ROOT_PATH = "/";
export const HISTORY_PATH = "/history";

const Routes = () => (
  <Switch>
    <Route exact path={ROOT_PATH} component={Main} />
    <Route exact path={HISTORY_PATH} component={History} />
    <Route component={NotFound} />
  </Switch>
);

export default Routes;
