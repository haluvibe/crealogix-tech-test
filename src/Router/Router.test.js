import React from "react";
import { shallow } from "enzyme";
import Router from "./Router";

describe("Router", () => {
  it("should render correctly with no props", () => {
    const wrapper = shallow(<Router />);

    expect(wrapper).toMatchSnapshot();
  });
  it("should render the Routes component", () => {
    const wrapper = shallow(<Router />);

    expect(wrapper.find("Routes")).toBeTruthy();
  });
});
