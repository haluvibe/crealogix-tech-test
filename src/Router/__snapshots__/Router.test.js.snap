// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should render correctly with no props 1`] = `
<BrowserRouter>
  <Routes />
</BrowserRouter>
`;
